public/scripts/main.js

```diff
@@ -56,16 +56,60 @@ function replaceWithInput(type, index) {
   }
 }
 
+function validateFields(bookNumber){
+  let exitCode = false
+  for(let field of fields){
+    const domField = document.querySelector('#'+field+'Input'+bookNumber)
+    let colorThisDom = false
```

Comment:
I'm not seeing that the colorThisDom variable is necessary: in all of the cases where it's assigned a value, exitCode is assigned the same value.